### ISSUE TITLE
Background and foreground colors do not have a sufficient contrast ratio.

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#9633]
See: https://github.com/internetarchive/openlibrary/issues/9633#issuecomment-2254156138

https://pagespeed.web.dev/analysis/https-openlibrary-org-books-OL1088853M-Wizard's_first_rule/e58x2c9cvi?form_factor=mobile gives a mobile score of 90 as of 2024-12-19. 

One criteria is reports is:

> Background and foreground colors do not have a sufficient contrast ratio.

<img width="736" alt="Screenshot 2024-12-19 at 8 20 49 AM" src="https://github.com/user-attachments/assets/bc565b3e-0c16-481b-9a84-72b6a61f0eee" />

`<a href="#overview" data-ol-link-track="BookPageNav|Overview">Overview</a>`

<img width="864" alt="Screenshot 2024-12-19 at 8 14 48 AM" src="https://github.com/user-attachments/assets/b49b5d97-8273-4c1f-afcd-15445af642a4" />

**Comment by veronicamar** [comment]
From PageSpeed:
Low-contrast text is difficult or impossible for many users to read.

<img width="904" alt="Screenshot 2024-07-27 at 2 54 15 pm" src="https://github.com/user-attachments/assets/5ee88abc-f878-4081-b8f8-fa5a4224d65e">

Ensure all text elements have sufficient color contrast between the text in the foreground and background color behind it.

Success Criterion: Ensure color contrast of at least 4.5:1 for small text or 3:1 for large text, even if text is part of an image. Large text has been defined in the requirements as 18pt (24 CSS pixels) or 14pt bold (19 CSS pixels). Note: Elements found to have a 1:1 ratio are considered "incomplete" and require a manual review.

Color Contrast Analysis Tools: Below are links to tools that will analyze color contrast ratios against guidelines:

[axe DevTools Browser Extensions](https://www.deque.com/axe/devtools/?__hstc=213731083.dd682d1c2c7c1af9b2bcbece37bc1089.1722087669599.1722087669599.1722087669599.1&__hssc=213731083.5.1722087669600&__hsfp=1910549127) — Deque's enterprise product, axe DevTools HTML, enables development groups to integrate accessibility testing into existing automated testing processes and is used by customers in conjunction with Deque's [axe Monitor](https://www.deque.com/axe/monitor/?__hstc=213731083.dd682d1c2c7c1af9b2bcbece37bc1089.1722087669599.1722087669599.1722087669599.1&__hssc=213731083.5.1722087669600&__hsfp=1910549127) accessibility monitoring and reporting product. In addition to providing a code library for integration in many programming languages, axe DevTools also includes web accessibility analysis extensions for Google Chrome and Mozilla Firefox, enabling enterprise front-end developers to run a quick accessibility test at any time. Analysis tools return detailed information on accessibility violations and instructions to fix issues with including links to more in-depth knowledge.
[axe-core](https://www.deque.com/axe/?__hstc=213731083.dd682d1c2c7c1af9b2bcbece37bc1089.1722087669599.1722087669599.1722087669599.1&__hssc=213731083.5.1722087669600&__hsfp=1910549127) — The axe accessibility engine is an open-source JavaScript accessibility rules library that is fast, returns no false positive errors or duplicate results, and is available as a GitHub repository, browser plugin, or framework integration.

**Comment by kohantikanath** [comment]
Hi @veronicamar,

I’d like to address the low-contrast text issue. I’ll ensure all text elements meet the required color contrast ratios for accessibility. Please assign me to this task!

**Comment by cdrini** [comment]
Thanks @kohantikanath ! This one needs some design before it can be implemented.

@veronicamar would you be willing propose some designs we can make so this can meet the colour criteria? Maybe changes to the colour, or changes to the styling of this UI element?

**Comment by veronicamar** [comment]
Hi so I have looked at passable colours for this and with the following it passes the accessibility test:

White: #FFFFFF
Blue: #3A5B7D - this is from the current shade used but contrast increased

Please let me know if you need further support

**Comment by rebecca-shoptaw** [comment]
Hi @veronicamar, thanks for working on this! That color contrast looks great to me, but I worry it would be too similar to the non-selected hover behavior, which the lighter blue was supposed to distinguish from:

<img width="264" alt="Selected vs. unselected links" src="https://github.com/user-attachments/assets/78b2e80b-1801-4307-9d25-ba47d1a32fab">

Perhaps instead, if the un-selected color (#456f9f) passes the accessibility test, we could make the selected link that color, and the un-selected link another existing color from the codebase, i.e. perhaps a gray like so:

<img width="269" alt="Possible alternate link color scheme" src="https://github.com/user-attachments/assets/e6a6cc54-b007-4d2c-988d-be6bb2b9bc3a">

Or even better, we could try to style the un-selected link to look like the vanilla button style (since I know the gray background usually indicates a disabled button on the site): 

<img width="67" alt="Sample vanilla button" src="https://github.com/user-attachments/assets/571be340-fa59-4e62-a5ab-7dfe34c42da8">

**Comment by veronicamar** [comment]
@rebecca-shoptaw I think those suggestions are great and you are totally right about the colour! Perhaps the 'edit' button is the best way to go forward as it will pass the accessibility score however I will look at other buttons on the site to see if it makes sense

**Comment by rebecca-shoptaw** [comment]
@veronicamar Thank you and that sounds good, would love to get some more input from others (@cdrini ? more design people?) before making a call since it's a fairly big design decision.

**Comment by mekarpeles** [comment]
Let's make the selected option ~black (something close to #333). This should distract less from the other primary calls to action.

The hover state can be the same primary blue as elsewhere on the site `#0376b8`.

<img width="659" alt="Screenshot 2024-12-19 at 8 34 09 AM" src="https://github.com/user-attachments/assets/d8c31941-ea55-4316-851a-61f8904bb808" />

@veronicamar shall we open a PR with these values? If we want to tweak them, that should be relatively simple once we already have the PR open.

**Comment by veronicamar** [comment]
@mekarpeles a PR would make sense! Apologies these seemed to be put on pause- ready to action asap

**Comment by lokesh** [comment]
We are redoing our color variables/tokens in #13102 so any work on this narrow issue could be obsolete soon. One of the sub-tasks of #13102 is to do a _Contrast sweep against the new color tokens_.
